project/SparkBuild.scala

```diff
@@ -266,7 +266,7 @@ object SparkBuild extends PomBuild {
       .orElse(sys.props.get("java.home").map { p => new File(p).getParentFile().getAbsolutePath() })
       .map(file),
     publishMavenStyle := true,
-    unidocGenjavadocVersion := "0.18",
+    unidocGenjavadocVersion := "0.19",
```

Comment:
Please upgrade `GenJavadoc ` in a separate pr first. @panbingkun

Reply:
Done

Reply:
@LuciferYang 
I think we can upgrade scala to `2.13.14` only after https://github.com/apache/spark/pull/46579 is merged.
1.
https://mvnrepository.com/artifact/com.typesafe.genjavadoc/genjavadoc-plugin
<img width="800" alt="image" src="https://github.com/apache/spark/assets/15246973/55b4e5c2-dd81-4f93-a301-c13bc06cce92">
<img width="807" alt="image" src="https://github.com/apache/spark/assets/15246973/f77fb9ae-1adf-4ddb-95a8-24951f9ff6c8">

2.
https://github.com/panbingkun/spark/actions/runs/9079936342/job/24951167973
<img width="972" alt="image" src="https://github.com/apache/spark/assets/15246973/d2c894c5-a5f1-4129-bbf4-e5720f41436c">